src/riakc_ts_query_operator.erl

```diff
@@ -31,9 +31,12 @@
 -export([serialize/2,
          deserialize/1, deserialize/2]).
 
-serialize(QueryText, Interpolations) ->
+
+serialize(QueryText, Interpolations)
+  when is_binary(QueryText) orelse is_list(QueryText) ->
+    Q = unicode:characters_to_binary(QueryText),
```

Comment:
To me it still has the air of crashing on user input (which is different from passing a `QueryText` of inappropriate type and crashing with a function_clause error). But I can convince myself that I'm wrong on that in this case.